### PR TITLE
TEAMNADO-4772 - Abstract common logic

### PR DESCRIPTION
**Pull request by lukegrn**
# Description

Associated Jira ticket: TEAMNADO-4772

This moves some common gating logic to a single protected button component

# Checklist:

- [x] The commit message has the Jira ticket linked
- [x] PR has a short description

### CODE REVIEW

File: src/Components/ActivationKey/EditAdditionalRepositoriesButton.js [hunk 1]
```diff
@@ -1,31 +1,23 @@
...
   );
 };
 
-EditAdditionalRepositoriesButton.propTypes = {
+AddAdditionalRepositoriesButton.propTypes = {
   onClick: PropTypes.func.isRequired,
 };
 
-export default EditAdditionalRepositoriesButton;
+export default AddAdditionalRepositoriesButton;
```
Comment (cowanrc) [inline src/Components/ActivationKey/EditAdditionalRepositoriesButton.js]: If we're changing everything in this from "Edit" to "Add" in every case here, should the file name also be changed to "AddAdditionalRepositoriesButton.js"?

Comment (lukegrn) [reply src/Components/ActivationKey/EditAdditionalRepositoriesButton.js]: Definitely - I thought I did. Whoops

Comment (lukegrn) [reply src/Components/ActivationKey/EditAdditionalRepositoriesButton.js]: Updated

**Review by cowanrc — APPROVED** [review]
LGTM

**Pull request merged**